### PR TITLE
adding api for pay direct transfers

### PR DESCRIPTION
**Pull request by IDubuque**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
The focus of this PR is to enhance the Pay feature by adding support for direct transfers in Pay. 

### Detailed summary
- Added support for direct transfers in Pay
- Updated `definitions.ts`, `getStatus.ts`, `BuyScreen.tsx`, `pay.ts`, and other relevant files
- Modified `commonTypes.ts` to include new transfer-related types
- Introduced `getBuyWithCryptoTransfer` function in `getTransfer.ts`
- Updated `direct-payment.tsx` and `transaction-button.tsx` for transfer-related changes

> The following files were skipped due to too many changes: `packages/thirdweb/src/pay/buyWithCrypto/getTransfer.ts`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/swap/TransferFlow.tsx`, `packages/thirdweb/src/react/web/ui/ConnectWallet/screens/Buy/swap/TransferConfirmationScreen.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by joaquim-verges** [comment]
* **#4224** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/4224?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @IDubuque and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/thirdweb/src/pay/buyWithCrypto/getTransfer.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  paymentToken: QuotePaymentToken;
+  processingFee: QuotePaymentToken;
+  estimatedGasCostUSDCents: number;
+  client: ThirdwebClient;
+};
+
+/**
+ * @buyCrypto
+ */
+export async function getBuyWithCryptoTransfer(
```
Comment (joaquim-verges) [inline packages/thirdweb/src/pay/buyWithCrypto/getTransfer.ts:97]: Note: needs tsdoc and added to exports

File: packages/thirdweb/src/utils/fetch.ts [hunk 2]
```diff
@@ -94,6 +94,7 @@ const THIRDWEB_DOMAINS = [
   // dev domains
   ".thirdweb.dev",
   ".thirdweb-dev.com",
+  ".chainsaw-dev.zeet.app",
```
Comment (joaquim-verges) [inline packages/thirdweb/src/utils/fetch.ts:97]: Remove this

File: apps/playground-web/src/components/pay/direct-payment.tsx [hunk 3]
```diff
@@ -16,9 +16,9 @@ export function BuyMerchPreview() {
         payOptions={{
           mode: "direct_payment",
           paymentInfo: {
-            amount: "15",
-            chain: base,
-            token: getDefaultToken(base, "USDC"),
+            amount: "0.1",
+            chain: sepolia,
+            token: getDefaultToken(sepolia, "USDC"),
             sellerAddress: "0xEb0effdFB4dC5b3d5d3aC6ce29F3ED213E95d675",
```
Comment (jnsdls) [inline apps/playground-web/src/components/pay/direct-payment.tsx:22]: was this meant to stay changed?

File: apps/playground-web/src/components/pay/transaction-button.tsx [hunk 4]
```diff
@@ -21,8 +22,9 @@ const nftContract = getContract({
 });
 
 const usdcContract = getContract({
-  address: "0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913",
-  chain: base,
+  // biome-ignore lint/style/noNonNullAssertion: its there
+  address: getDefaultToken(sepolia, "USDC")!.address,
+  chain: sepolia,
```
Comment (jnsdls) [inline apps/playground-web/src/components/pay/transaction-button.tsx:27]: meant to change this? also if its there then the type should correctly infer that, right?

File: apps/playground-web/src/components/pay/direct-payment.tsx (hunk 3 above)
Comment (joaquim-verges) [reply apps/playground-web/src/components/pay/direct-payment.tsx:22]: Yes. Easier to test for ppl. Don't really wanna charge real money

File: apps/playground-web/src/components/pay/transaction-button.tsx (hunk 4 above)
Comment (joaquim-verges) [reply apps/playground-web/src/components/pay/transaction-button.tsx:27]: Also yes. Function is a bit loose right now, not hard typed. Fine for now tho

**Review by joaquim-verges — APPROVED** [review]

**Pull request merged**